### PR TITLE
Pass translation_classes_data to sumpy

### PR DESCRIPTION
**Pull request by isuruf**
When this is passed to sumpy, it will use the optimized code path where the M2L matrix is precomputed.

### CODE REVIEW

**Comment by inducer** [comment]
Marking draft and unsubscribing for CI fails, let me know (request review/@-mention) when ready.

**Comment by isuruf** [comment]
@inducer, this is ready.

File: pytential/qbx/__init__.py [hunk 1]
```diff
@@ -603,6 +603,9 @@ def exec_compute_potential_insn_fmm(self, actx: PyOpenCLArrayContext,
                     geo_data.tree().user_source_ids,
                     insn.kernel_arguments, evaluate))
 
+        from sumpy.fmm import SumpyTranslationClassesData
+        translation_classes_data = SumpyTranslationClassesData(actx.queue,
+                geo_data.traversal())
```
Comment (inducer) [inline pytential/qbx/__init__.py:608]: This isn't a sumpy-specific interface. How come we're passing ostensibly sumpy-specific data?

File: pytential/qbx/__init__.py [hunk 2]
```diff
@@ -603,6 +603,9 @@ def exec_compute_potential_insn_fmm(self, actx: PyOpenCLArrayContext,
...
         tree_indep = self._tree_indep_data_for_wrangler(
```
Comment (inducer) [inline pytential/qbx/__init__.py:609]: Would it make sense to compute the translation classes as part of the wrangler creation?

**Review by inducer — COMMENTED** [review]
Thanks for working on this. Two questions below.

File: pytential/qbx/__init__.py (hunk 2 above)
Comment (isuruf) [reply pytential/qbx/__init__.py:609]: Wrangler doesn't have access to a `queue`. Should I add that?

Comment (inducer) [reply pytential/qbx/__init__.py:609]: You can just make a private, temporary queue:
```python
with cl.CommandQueue(ctx) as queue:
   ...
```

**Comment by isuruf** [comment]
Made redundant by https://github.com/inducer/sumpy/pull/117

**Pull request closed without merging**